pkg/operator/observe_config.go

```diff
@@ -172,23 +172,39 @@ func observeBuildControllerConfig(listers Listers, observedConfig map[string]int
 		return nil, err
 	}
 	// set build defaults
-	if len(build.Spec.BuildDefaults.GitHTTPProxy) > 0 {
-		if err = observeField(observedConfig, build.Spec.BuildDefaults.GitHTTPProxy, "build.buildDefaults.gitHTTPProxy", true); err != nil {
+
+	if build.Spec.BuildDefaults.GitProxy != nil {
+		if err = observeField(observedConfig, build.Spec.BuildDefaults.GitProxy.HTTPProxy, "build.buildDefaults.gitHTTPProxy", false); err != nil {
 			return nil, fmt.Errorf("failed to observe %s: %v", "build.buildDefaults.gitHTTPProxy", err)
 		}
-	}
-	if len(build.Spec.BuildDefaults.GitHTTPSProxy) > 0 {
-		if err = observeField(observedConfig, build.Spec.BuildDefaults.GitHTTPSProxy, "build.buildDefaults.gitHTTPSProxy", true); err != nil {
+		if err = observeField(observedConfig, build.Spec.BuildDefaults.GitProxy.HTTPSProxy, "build.buildDefaults.gitHTTPSProxy", false); err != nil {
 			return nil, fmt.Errorf("failed to observe %s: %v", "build.buildDefaults.gitHTTPSProxy", err)
 		}
-	}
-	if len(build.Spec.BuildDefaults.GitNoProxy) > 0 {
-		if err = observeField(observedConfig, build.Spec.BuildDefaults.GitNoProxy, "build.buildDefaults.gitNoProxy", true); err != nil {
+		if err = observeField(observedConfig, build.Spec.BuildDefaults.GitProxy.NoProxy, "build.buildDefaults.gitNoProxy", false); err != nil {
 			return nil, fmt.Errorf("failed to observe %s: %v", "build.buildDefaults.gitNoProxy", err)
 		}
 	}
-	if len(build.Spec.BuildDefaults.Env) > 0 {
-		if err = observeField(observedConfig, build.Spec.BuildDefaults.Env, "build.buildDefaults.env", true); err != nil {
+
+	buildEnv := build.Spec.BuildDefaults.Env
+	// Translate DefaultProxy valuse into env vars used by openshift/builder
+	if build.Spec.BuildDefaults.DefaultProxy != nil {
+		buildEnv = append(buildEnv, corev1.EnvVar{
+			Name:  "HTTP_PROXY",
+			Value: build.Spec.BuildDefaults.DefaultProxy.HTTPProxy,
+		})
+		buildEnv = append(buildEnv, corev1.EnvVar{
+			Name:  "HTTPS_PROXY",
+			Value: build.Spec.BuildDefaults.DefaultProxy.HTTPSProxy,
+		})
+		buildEnv = append(buildEnv, corev1.EnvVar{
+			Name:  "NO_PROXY",
+			Value: build.Spec.BuildDefaults.DefaultProxy.NoProxy,
+		})
+	}
```

Comment:
this needs to be handled differently.  builddefaults.env is for env variables that are supposed to be added to the application image that is built.  it is not for env vars to be added to the build pod (more importantly, it's likely undesirable to do this because these vars will end up in the application image).

please limit this PR to fixing up the gitproxy defaulting part for now as the correct solution to the defaultproxy is going to require more pieces.

Reply:
@bparees removed.